packages/client/src/runtime/core/request/RequestPromise.ts

```diff
@@ -0,0 +1,66 @@
+import { IsolationLevel } from '@prisma/engine-core'
+import { Optional } from 'ts-toolbelt/out/Object/Optional'
+
+import { ExtensionOptions } from '../extension/$extends'
+
+export type RequestPromiseTransactionOptions = {
+  transaction:
+    | {
+        kind: 'itx'
+        id: string
+      }
+    | {
+        kind: 'batch'
+        id: number
+        isolationLevel?: IsolationLevel
+        lock?: PromiseLike<void>
+      }
+}
+
+export type RequestPromiseExtensionOptions = {
+  extensions: ExtensionOptions[]
+}
+
+export type RequestPromiseOptions = Optional<RequestPromiseTransactionOptions & RequestPromiseExtensionOptions>
+
+/**
+ * Prisma's `Promise` that is backwards-compatible. All additions on top of the
+ * original `Promise` are optional so that it can be backwards-compatible.
+ * @see [[createRequestPromise]]
+ */
+export interface RequestPromise<A> extends Promise<A> {
+  /**
+   * Extension of the original `.then` function
+   * @param onfulfilled same as regular promises
+   * @param onrejected same as regular promises
+   * @param options execution options
+   */
+  then<R1 = A, R2 = never>(
+    onfulfilled?: (value: A) => R1 | PromiseLike<R1>,
+    onrejected?: (error: unknown) => R2 | PromiseLike<R2>,
+    options?: RequestPromiseOptions,
+  ): Promise<R1 | R2>
+
+  /**
+   * Extension of the original `.catch` function
+   * @param onrejected same as regular promises
+   * @param options execution  options
+   */
+  catch<R = never>(
+    onrejected?: ((reason: any) => R | PromiseLike<R>) | undefined | null,
+    options?: RequestPromiseOptions,
+  ): Promise<A | R>
+
+  /**
+   * Extension of the original `.finally` function
+   * @param onfinally same as regular promises
+   * @param options execution options
+   */
+  finally(onfinally?: (() => void) | undefined | null, options?: RequestPromiseOptions): Promise<A>
+
+  /**
+   * Called when executing a batch of regular tx
+   * @param options execution options
+   */
+  requestTransaction?(options?: RequestPromiseOptions): PromiseLike<unknown>
```

Comment:
Would `requestTransaction` call make sense for `{ kind: 'itx'}`? Original code assumed it won't and made `requestTransaction` explicitly accept only batch options.